sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -4061,10 +4061,30 @@ object SessionWindowing extends Rule[LogicalPlan] {
           case s: SessionWindow => sessionAttr
         }
 
+        def isGapInvalid(child: Expression): Boolean = {
+          val calendarInterval = IntervalUtils
+            .safeStringToInterval(UTF8String.fromString(child.toString))
+          calendarInterval == null ||
+            calendarInterval.months + calendarInterval.days + calendarInterval.microseconds <= 0
+        }
+
+        val filterTimeSize = gapDuration match {
+          case gapDuration if gapDuration.find(_.isInstanceOf[ScalaUDF]).nonEmpty => true
+          case _ => gapDuration.child.children match {
+            case children if children.size > 0 =>
+              children.filter(_.dataType != BooleanType).exists(isGapInvalid(_))
+            case _ => isGapInvalid(gapDuration.child)
+          }
```

Comment:
Seems we only need to check if the (casted) gap duration expression is foldable. If so, just evaluating it to get Calendar Interval value?

Reply:
Excluding `UDF` and `static gap`, it seems like this. Filter out `BooleanType`, which is calendar interval value.